src/hooks/useQuery.ts

```diff
@@ -39,27 +40,34 @@ export const useQuery = <T = any, V = object>(
   const executeQuery = useCallback(
     (opts?: Partial<OperationContext>) => {
       unsubscribe();
+
       setState(s => ({ ...s, fetching: true }));
 
-      const [teardown] = pipe(
-        client.executeQuery(request, {
-          requestPolicy: args.requestPolicy,
-          ...opts,
-        }),
-        subscribe(({ data, error }) =>
-          setState({ fetching: false, data, error })
-        )
-      );
+      let teardown = noop;
+
+      if (!args.skip) {
+        [teardown] = pipe(
+          client.executeQuery(request, {
+            requestPolicy: args.requestPolicy,
```

Comment:
Looking at this now, shouldn't `args.requestPolicy` also be included in the dependency array for this `useCallback`? Otherwise we might not recompute it if it changes.

Reply:
Yup I believe so. We're currently allowing users to specify the request policy on a manual query execution but I believe this goes against the way hooks work.

If the request policy changes, that should be requested in the components state and flow all the way down to the useQuery hook.

Reply:
Pushed!